notebooks/train_and_infer.ipynb

```diff
@@ -2,19 +2,161 @@
  "cells": [
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #53.        #pos = {k: (pos[k][1], pos[k][0]) for k in pos}
</pre><p>should we remove this code?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/alan-turing-institute/grace/pull/92/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>